torat_client/rpcapi.go

```diff
@@ -71,9 +72,23 @@ func (a *API) LS(v shared.Void, r *shared.Dir) (err error) {
 	return
 }
 
-func (a *API) Ping(v shared.Void, r *string) error {
-	// TODO implement
-	*r = "Pong"
+func (a *API) Speedtest(v shared.Void, r *shared.Speedtest) error {
+	user, _ := speedtest.FetchUserInfo()
```

Comment:
You need to check the error here otherwise line 85 might throw a nil pointer panic on accessing the IP field.

Please also check the other errors while at it (line 77 and 78)

Reply:
Before adding all the error handling let me do some testing. I don't think it will panic, just return an empty struct. Plus the official example in the docs for this module just throws away the error.